src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/Utils.scala

```diff
@@ -55,4 +62,80 @@ object Utils {
       tenant.attributes.mapFilter(_.certified.map(_.id)) ++
       tenant.attributes.mapFilter(_.declared.map(_.id))
 
+  def isUpgradable(
+    descriptor: CatalogManagement.EServiceDescriptor,
+    descriptors: Seq[CatalogManagement.EServiceDescriptor]
+  ): Boolean =
+    descriptors
+      .filter(_.version.toInt > descriptor.version.toInt)
+      .exists(d =>
+        d.state == CatalogManagement.EServiceDescriptorState.PUBLISHED ||
+          d.state == CatalogManagement.EServiceDescriptorState.SUSPENDED
+      )
+
+  def isUpgradable(
+    descriptor: CatalogProcess.EServiceDescriptor,
+    descriptors: Seq[CatalogProcess.EServiceDescriptor]
+  ): Boolean =
+    descriptors
+      .filter(_.version.toInt > descriptor.version.toInt)
+      .exists(d =>
+        d.state == CatalogProcess.EServiceDescriptorState.PUBLISHED ||
+          d.state == CatalogProcess.EServiceDescriptorState.SUSPENDED
+      )
+
+  def getAllAgreements(
```

Comment:
isn't this used only in the `EServicesApiServiceImpl`?
if yes, we do not need to move it

Reply:
I agree with @galales

Reply:
Why not move this to the AgreementProcess trait?